platform/machine/packet/flight.go

```diff
@@ -94,4 +99,10 @@ func (pf *flight) Destroy() {
 	}
 
 	pf.BaseFlight.Destroy()
+
+	for id := range pf.devicesPool {
```

Comment:
```suggestion
	close(pf.devicesPool)
	for id := range pf.devicesPool {
```
Otherwise it will deadlock and wait endlessly.
The task here is to delete all devices that got created. This function is called when all tests are done and no new machines get created but we should close the channel to make sure that this is the case (instead of working around the deadlock with a possible select or similar).